src/state_point.F90

```diff
@@ -238,59 +276,108 @@ subroutine write_state_point()
             call close_group(filter_group)
           end do FILTER_LOOP
 
-          call write_dataset(tally_group, "n_nuclides", tally%n_nuclide_bins)
-
           ! Set up nuclide bin array and then write
-          allocate(key_array(tally%n_nuclide_bins))
+          allocate(str_array(tally%n_nuclide_bins))
           NUCLIDE_LOOP: do j = 1, tally%n_nuclide_bins
             if (tally%nuclide_bins(j) > 0) then
-              key_array(j) = nuclides(tally%nuclide_bins(j))%zaid
+              i_list = nuclides(tally%nuclide_bins(j))%listing
+              i_xs = index(xs_listings(i_list)%alias, '.')
+              if (i_xs > 0) then
```

Comment:
It might be nice to include a comment around this ``if-else`` to clarify what ``i_xs`` is doing here.